gpytorch/priors/lkj_prior.py

```diff
@@ -130,6 +175,22 @@ def log_prob(self, X):
         log_prob_sd = self.sd_prior.log_prob(marginal_sd)
         return log_prob_corr + log_prob_sd
 
+    def rsample(self, sample_shape):
+        base_correlation = self.correlation_prior.rsample(sample_shape)
+        marginal_sds = self.sd_prior.rsample(sample_shape)
+        # create an empty diagonal matrix
+        empty_diag_mat = torch.zeros(
+            *sample_shape,
+            self.correlation_prior.n,
+            self.correlation_prior.n,
+            device=base_correlation.device,
+            dtype=base_correlation.dtype,
+        )
+        # and add to the batch dimension
+        marginal_sds = marginal_sds.unsqueeze(-1) + empty_diag_mat
```

Comment:
What exactly are you trying to do here? I don't think `marginal_sds` will be diagonal after this. See comment above how to implement a covar-to-corr or corr-to-covar scaling more efficiently by avoiding matmuls.

Reply:
I'm trying to embed the marginal sds in as a diagonal. Hopefully this isn't too boneheaded.